### DISCUSSION TITLE
How to access an object in a map layer programmatically

### DISCUSSION THREAD

**Discussion by convers39** [visgl/deck.gl#7312] [Q&A]
I am doing E2E test on a React app with Cypress, and trying to access the objects rendered on the GeoJsonLayer.

I wonder if it is possible to retrieve the x, y position of the objects in the layer, so that I can simulate with Cypress to click on the position. Or is there any other way to simulate a click event for an object in a layer?

I understand there are no corresponding DOM elements for those objects, so have no idea how to programmatically access the objects.

```ts
import { StaticMap } from "react-map-gl"
import DeckGL from "@deck.gl/react"
import { GeoJsonLayer } from "@deck.gl/layers"


const layers = new GeoJsonLayer({
    id: '',
    data: dataList.map(d => d.geoJson)
    pickable:true,
    autoHighlight: true,
    getLineColor: [0, 0, 255],
    getLineWidth: 6,
    lineWidthMinPixels: 1,
    onClick: (info) => {
        const { x, y, object, coordinate } = info;
        // ... 
    },

})
return (
  <DeckGL
      onViewStateChange={onViewStateChange}
      viewState={viewState}
      layers={layers}
      controller={true}
     >
     <StaticMap
          width={"100%"}
          height={"100%"}
          mapStyle={`mapbox://styles/mapbox/${mapStyle}`}
          mapboxApiAccessToken={MAPBOX_ACCESS_TOKEN}
      />
  </DeckGL>

)
```

**Comment by Pessimistress** [comment]
You can use [WebMercatorViewport](https://deck.gl/docs/api-reference/core/web-mercator-viewport) to convert a geo coordinate to screen coordinates.
